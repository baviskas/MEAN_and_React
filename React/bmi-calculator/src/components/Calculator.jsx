import React, { Component } from 'react';
import AttributesInput from './AttributesInput';
import BMIVerdict from './BMIVerdict';
import { kgToPound, poundToKg, tryConvert, calculateBMI }  from './../util/util';

class Calculator extends Component {
    constructor(props) {
        super(props);
        this.handleKGChange = this.handleKGChange.bind(this);
        this.handlePoundChange = this.handlePoundChange.bind(this);
        this.handleHeightChange = this.handleHeightChange.bind(this);
        this.state = {
            weight: '',
            height: '',
            weightScale: 'k'
        };
    }

    handleKGChange(weight) {
        this.setState({ weight, weightScale: 'k' });
    }

    handlePoundChange(weight) {
        this.setState({ weight, weightScale: 'p' });
    }

    handleHeightChange(height) {
        this.setState({height});
    }
    render() {
        const scale = this.state.weightScale;
        const weight = this.state.weight;
        const height = this.state.height;
        const kg = scale === 'p' ? tryConvert(weight, poundToKg) : weight;
        const pound = scale === 'k' ? tryConvert(weight, kgToPound) : weight;
        const BMI = calculateBMI(parseFloat(height),kg);
        return (
            <div>
                <h2>Please enter your weight & height: </h2>
                <AttributesInput
                    attribute={'Weight'}
                    scale={'k'}
                    value={kg}
                    onInputChange={this.handleKGChange}
                />
                <AttributesInput
                    attribute={'Weight'}
                    scale={'p'}
                    value={pound}
                    onInputChange={this.handlePoundChange}
                />
                <AttributesInput
                    attribute={'Height'}
                    scale={'m'}
                    value={height}
                    onInputChange={this.handleHeightChange}
                />
                <BMIVerdict BMI={BMI}/>
            </div>
        )
    }
}

export default Calculator;
