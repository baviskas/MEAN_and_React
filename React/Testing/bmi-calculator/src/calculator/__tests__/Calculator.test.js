import { shallowWrapper } from "./../../testUtil";
import Calculator from "../Calculator";

const requiredProps = {};
const wrapComponent = (props) => {
    return shallowWrapper(Calculator, requiredProps, props);
};

describe("Calculator: ", () => {
    afterEach(() => {
        jest.clearAllMocks();
    });

    describe("Smoke Test", () => {
        it("render smoketest", () => {
            const wrapper = wrapComponent();
            expect(wrapper.debug()).toMatchSnapshot();
        });
    });

    describe("Test methods", () => {
       it("sets weight in state in kg", () => {
           const wrapper = wrapComponent();
           wrapper.instance().handleWeightInKg(100);
           expect(wrapper.instance().state.weight).toEqual(100);
           expect(wrapper.instance().state.weightScale).toEqual("k");
       });

        it("sets weight in state in pounds", () => {
            const wrapper = wrapComponent();
            wrapper.instance().handleWeightInPounds(100);
            expect(wrapper.instance().state.weight).toEqual(100);
            expect(wrapper.instance().state.weightScale).toEqual("p");
        });

        it("sets height in state", () => {
            const wrapper = wrapComponent();
            wrapper.instance().handleHeight(1.80);
            expect(wrapper.instance().state.height).toEqual(1.80);
        });
    });
});
