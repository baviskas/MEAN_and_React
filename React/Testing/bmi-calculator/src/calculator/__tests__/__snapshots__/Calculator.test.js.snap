// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator:  Smoke Test render smoketest 1`] = `
"<Fragment>
  <h1>
    BMI Calculator:
  </h1>
  <AttributesInput value=\\"\\" attribute=\\"Weight\\" scale=\\"k\\" handler={[Function: bound handleWeightInKg]} />
  <AttributesInput value=\\"\\" attribute=\\"Weight\\" scale=\\"p\\" handler={[Function: bound handleWeightInPounds]} />
  <AttributesInput value=\\"\\" attribute=\\"Height\\" scale=\\"m\\" handler={[Function: bound handleHeight]} />
  <BMIVerdict bmi={NaN} />
</Fragment>"
`;

exports[`VinInput:  Smoke Test render smoketest 1`] = `
"<Fragment>
  <h1>
    BMI Calculator:
  </h1>
  <AttributesInput value=\\"\\" attribute=\\"Weight\\" scale=\\"k\\" handler={[Function: bound handleWeightInKg]} />
  <AttributesInput value=\\"\\" attribute=\\"Weight\\" scale=\\"p\\" handler={[Function: bound handleWeightInPounds]} />
  <AttributesInput value=\\"\\" attribute=\\"Height\\" scale=\\"m\\" handler={[Function: bound handleHeight]} />
  <BMIVerdict bmi={NaN} />
</Fragment>"
`;
